src/com/goide/runconfig/testing/GoTestRunningState.java

```diff
@@ -83,13 +84,17 @@ protected GoExecutor patchExecutor(@NotNull GoExecutor executor) throws Executio
         String relativePath = FileUtil.getRelativePath(myConfiguration.getWorkingDirectory(),
                                                        myConfiguration.getDirectoryPath(),
                                                        File.separatorChar);
+        String path = "." + File.separator;
         if (relativePath != null) {
-          executor.withParameters("./" + relativePath + "/...");
+          path += relativePath;
         }
         else {
-          executor.withParameters("./...");
           executor.withWorkDirectory(myConfiguration.getDirectoryPath());
         }
+        // TODO Once Go gets support for covering multiple packages the ternary condition should be reverted
+        // See https://golang.org/issues/6909
+        if (!path.endsWith(File.separator)) path += File.separator;
```

Comment:
Now flow is overcomplicated. Path can end with separator only if relativePath is not null and appending it with File.separator makes sense only if it's not coverage run. As a result you make 3 ifs instead 2.

The only thing you change is path suffix ("..." or "."), you can extract it in variable and append it to executor for both cases (with null relativePath and notnull relativePath)